### PR TITLE
Add inference tutorials for third-party models

### PR DESCRIPTION
**Pull request by VictorHe-1**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

Improve inference docs

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: docs/cn/inference/models_list_thirdparty.md [hunk 1]
```diff
@@ -1,11 +1,134 @@
...
+    C --input --> D[infer.py] -- outputs --> eval_rec.py/eval_det.py;
+    H[images] --input --> D[infer.py];
+```
 
-### 1. 文本检测
+### 2. 第三方模型推理方法
+#### 文本检测
+下面以[附录表格](#2-附录-第三方模型支持列表)中的`en_pp_det_dbnet_resnet50vd`为例介绍推理方法：
+- 下载附录表格中的权重文件[weight](https://paddleocr.bj.bcebos.com/dygraph_v2.0/en/det_r50_vd_db_v2.0_train.tar)并解压；
+- 根据[模型转换教程](convert_tutorial.md)，由于该模型为paddle训练模型，需要先转换为推理模型（已为推理模型则跳过此步）：
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:14]:
```suggestion
- 由于该模型为paddle训练模型，需要先转换为推理模型（已为推理模型则跳过此步）：
```

File: docs/cn/inference/models_list_thirdparty.md [hunk 2]
```diff
@@ -1,11 +1,134 @@
...
+```shell
+git clone https://github.com/PaddlePaddle/PaddleOCR.git
+cd PaddleOCR
+python tools/export_model.py \
+	-c configs/det/det_r50_vd_db.yml \
+	-o Global.pretrained_model=./det_r50_vd_db_v2.0_train/best_accuracy  \
+	Global.save_inference_dir=./det_db
+```
+执行完成后会生成`det_db`文件夹，包含`inference.pdiparams`，`inference.pdiparams.info`，`inference.pdmodel`文件；
+- 下载并使用paddle2onnx工具(`pip install paddle2onnx==0.9.5`)，将推理模型转换成onnx文件：
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:24]:
```suggestion
> 了解更多[模型转换教程](convert_tutorial.md)
- 下载并使用paddle2onnx工具(`pip install paddle2onnx==0.9.5`)，将推理模型转换成onnx文件：
```

File: docs/cn/inference/models_list_thirdparty.md [hunk 3]
```diff
@@ -1,11 +1,134 @@
...
+- 根据[模型转换教程](convert_tutorial.md)，由于该模型为paddle训练模型，需要先转换为推理模型（已为推理模型则跳过此步）：
+```shell
+git clone https://github.com/PaddlePaddle/PaddleOCR.git
+cd PaddleOCR
+python tools/export_model.py \
+	-c configs/det/det_r50_vd_db.yml \
+	-o Global.pretrained_model=./det_r50_vd_db_v2.0_train/best_accuracy  \
+	Global.save_inference_dir=./det_db
+```
+执行完成后会生成`det_db`文件夹，包含`inference.pdiparams`，`inference.pdiparams.info`，`inference.pdmodel`文件；
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:23]: 用文件夹tree的形式展现。
```
det_db
  inference.pdiparams
  ...
```

File: docs/cn/inference/models_list_thirdparty.md [hunk 4]
```diff
@@ -1,11 +1,134 @@
...
+converter_lite \
+    --saveType=MINDIR \
+    --NoFusion=false \
+    --fmk=ONNX \
+    --device=Ascend \
+    --modelFile=det_db.onnx \
+    --outputFile=det_db_output \
+    --configFile=config.txt
+```
+上述命令执行完成后会生成`det_db_output.mindir`和`det_db_output.om`文件，其中`config.txt`需要提前创建并指定模型输入shape，一个示例如下：
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:48]:
```suggestion
其中`config.txt`需要提前创建并指定模型输入shape，一个示例如下：
```

File: docs/cn/inference/models_list_thirdparty.md [hunk 5]
```diff
@@ -1,11 +1,134 @@
...
+```
+[ascend_context]
+input_format=NCHW
+input_shape=x:[1,3,736,1280]
+```
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:53]: 说完config.txt的配置后，再说“上述命令执行完成后会生成`det_db_output.mindir`和`det_db_output.om`文件，”。用文件夹tree的形式展现目前文件夹里都有什么文件了。

File: docs/cn/inference/models_list_thirdparty.md [hunk 6]
```diff
@@ -1,11 +1,134 @@
...
+    --input_images_dir=/path/to/mlt17_en \
+    —-device_id=1 \
+    —-parallel_num=2 \
+    —-rec_model_path=/path/to/mindir/en_PP-OCRv3_rec_infer.mindir \
+    —-rec_model_name_or_config=en_pp_rec_OCRv3 \
+    —-character_dict_path=/path/to/en_dict.txt \
+    --res_save_dir=/path/to/en_rec_infer_results \
+    —-save_log_dir=/path/to/en_rec_infer_logs
+```
+执行完成后，在`/path/to/en_rec_infer_results`目录下生成预测文件`rec_results.txt`。
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:120]:
```suggestion
执行完成后，在参数`--res_save_dir`所指目录下生成预测文件`det_results.txt`
```

File: docs/cn/inference/models_list_thirdparty.md [hunk 7]
```diff
@@ -1,11 +1,134 @@
...
+    --model_dir det_db \
+    --model_filename inference.pdmodel \
+    --params_filename inference.pdiparams \
+    --save_file det_db.onnx \
+    --opset_version 11 \
+    --input_shape_dict="{'x':[-1,3,-1,-1]}" \
+    --enable_onnx_checker True
+```
 
+参数中`input_shape_dict`的值，一般可以通过[Netron](https://github.com/lutzroeder/netron)工具打开推理模型查看，或者在上述[tools/export_model.py](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/tools/export_model.py)的代码中找到，上述命令执行完成后会生成`det_db.onnx`文件;
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:36]:
```suggestion
参数中`input_shape_dict`的值，一般可以通过[Netron](https://github.com/lutzroeder/netron)工具打开推理模型查看，或者在上述[tools/export_model.py](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/tools/export_model.py)的代码中找到。

上述命令执行完成后会生成`det_db.onnx`文件;
```

File: docs/cn/inference/models_list_thirdparty.md [hunk 8]
```diff
@@ -1,11 +1,134 @@
...
+- 在Ascend310/310P上使用[converter_lite](https://www.mindspore.cn/lite/docs/zh-CN/master/use/cloud_infer/converter_tool.html)工具将onnx文件转换为mindir：
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:37]:
```suggestion
- 在Ascend310/310P上使用converter_lite工具将onnx文件转换为mindir：
```

File: docs/cn/inference/models_list_thirdparty.md (hunk 2 above)
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:24]:
```suggestion
- 下载并使用paddle2onnx工具(`pip install paddle2onnx`)，将推理模型转换成onnx文件：
```

File: docs/cn/inference/models_list_thirdparty.md [hunk 9]
```diff
@@ -1,11 +1,134 @@
...
+```
+[ascend_context]
+input_format=NCHW
+input_shape=x:[1,3,736,1280]
+```
+- 使用`/deploy/py_infer/infer.py`脚本和`det_db_output.mindir`文件执行推理：
+```shell
+python infer.py \
+    --input_images_dir=/path/to/ic15/ch4_test_images \
+    —-device_id=1 \
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:58]: 删去不必要参数

File: docs/cn/inference/models_list_thirdparty.md (hunk 1 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 3 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 4 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 5 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 6 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md [hunk 10]
```diff
@@ -1,11 +1,134 @@
...
+    --outputFile=det_db_output \
+    --configFile=config.txt
+```
+上述命令执行完成后会生成`det_db_output.mindir`和`det_db_output.om`文件，其中`config.txt`需要提前创建并指定模型输入shape，一个示例如下：
+```
+[ascend_context]
+input_format=NCHW
+input_shape=x:[1,3,736,1280]
+```
+- 使用`/deploy/py_infer/infer.py`脚本和`det_db_output.mindir`文件执行推理：
```
[the remark this answers is not in the document]
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md:67]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 7 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 8 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 2 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md (hunk 9 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: 已修改

File: docs/cn/inference/models_list_thirdparty.md [hunk 11]
```diff
@@ -1,11 +1,151 @@
...
+paddle2onnx \
+    --model_dir en_PP-OCRv3_rec_infer \
+    --model_filename inference.pdmodel \
+    --params_filename inference.pdiparams \
+    --save_file en_PP-OCRv3_rec_infer.onnx \
+    --opset_version 11 \
+    --input_shape_dict="{'x':[-1,3,48,-1]}" \
+    --enable_onnx_checker True
+```
+参数中`input_shape_dict`的值，可以通过[Netron](https://github.com/lutzroeder/netron)工具打开推理模型查看，上述命令执行完成后会生成`en_PP-OCRv3_rec_infer.onnx`文件;
```
Comment (liangxhao) [inline docs/cn/inference/models_list_thirdparty.md:101]: --input_shape_dict

File: docs/cn/inference/models_list_thirdparty.md [hunk 12]
```diff
@@ -1,11 +1,151 @@
...
+git clone https://github.com/PaddlePaddle/PaddleOCR.git
+cd PaddleOCR
+python tools/export_model.py \
+	-c configs/det/det_r50_vd_db.yml \
+	-o Global.pretrained_model=./det_r50_vd_db_v2.0_train/best_accuracy  \
+	Global.save_inference_dir=./det_db
+```
+执行完成后会生成以下内容：
+```
+det_db
```
Comment (HaoyangLee) [inline docs/cn/inference/models_list_thirdparty.md:25]: 改进下tree格式

**Review by HaoyangLee — APPROVED** [review]

File: docs/cn/inference/models_list_thirdparty.md (hunk 11 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: ok

File: docs/cn/inference/models_list_thirdparty.md (hunk 12 above)
Comment (VictorHe-1) [reply docs/cn/inference/models_list_thirdparty.md]: ok

**Review by HaoyangLee — APPROVED** [review]

**Review by liangxhao — APPROVED** [review]

**Pull request merged**